realm/realm-library/src/main/java/io/realm/internal/ColumnInfo.java

```diff
@@ -217,39 +222,28 @@ public String toString() {
      * <p>
      * No validation done here.  Presuming that all necessary validation takes place in {@code Proxy.validateTable}.
      *
-     * @param table The table to search for the column.
      * @param columnName The name of the column whose index is sought.
-     * @param columnType Type RealmType of the column.
+     * @param objectSchemaInfo the {@link OsObjectSchemaInfo} for the corresponding {@code RealmObject}.
      * @return the index of the column in the table
      */
-    @SuppressWarnings("unused")
-    protected final long addColumnDetails(Table table, String columnName, RealmFieldType columnType) {
-        long columnIndex = table.getColumnIndex(columnName);
-        if (columnIndex >= 0) {
-            String linkedTableName = ((columnType != RealmFieldType.OBJECT) && (columnType != RealmFieldType.LIST))
-                    ? null
-                    : table.getLinkTarget(columnIndex).getClassName();
-
-            indicesMap.put(columnName, new ColumnDetails(columnIndex, columnType, linkedTableName));
-        }
-
-        return columnIndex;
+    protected final long addColumnDetails(String columnName, OsObjectSchemaInfo objectSchemaInfo) {
+        Property property = objectSchemaInfo.getProperty(columnName);
+        indicesMap.put(columnName, new ColumnDetails(property));
+        return property.getColumnIndex();
     }
 
     /**
      * Add a new backlink to the indexMap.
      * <b>For use only by subclasses!</b>.
      * Must be called from within the subclass constructor, to maintain the effectively-final contract.
      *
-     * @param realm The shared realm.
+     * @param schemaInfo the {@link OsSchemaInfo} of the corresponding {@code Realm} instance.
      * @param columnName The name of the backlink column.
      * @param sourceTableName The name of the backlink source class.
      * @param sourceColumnName The name of the backlink source field.
      */
-    @SuppressWarnings("unused")
```

Comment:
Doesn't need it anymore? It looks like invoking `addBacklinkDetails` are similar.